console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,90 @@ type ConsolePluginSpec struct {
+	//   ConsolePlugin A directives:
+	//     script-src: https://script1.com/ https://script2.com/
+	//     font-src: https://font1.com/
+	//
+	//   ConsolePlugin B directives:
+	//     script-src: https://script2.com/ https://script3.com/
+	//     font-src: https://font2.com/
+	//     img-src: https://img1.com/
+	//
+	//   OpenShift web console server CSP response header:
+	//     script-src: self https://script1.com/ https://script2.com/ https://script3.com/
+	//     font-src: self https://font1.com/ https://font2.com/
+	//     img-src: self https://img1.com/
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=5
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	ContentSecurityPolicy []ConsolePluginCSP `json:"contentSecurityPolicy,omitempty"`
+}
+
+// DirectiveType is an enumeration of OpenShift web console supported CSP directives.
+// LoadType is an enumeration of i18n loading types.
+// +kubebuilder:validation:Enum:=default-src;script-src;img-src;style-src;font-src
+type DirectiveType string
+
+const (
+	// default-src directive serves as a fallback for the other CSP fetch directives.
+	// For more information about the default-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src
+	DefaultSrc DirectiveType = "default-src"
+	// script-src directive specifies valid sources for JavaScript.
+	// For more information about the script-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src
+	ScriptSrc DirectiveType = "script-src"
+	// img-src directive specifies a valid sources of images and favicons.
+	// For more information about the img-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src
+	ImgSrc DirectiveType = "img-src"
+	// style-src directive specifies valid sources for stylesheets.
+	// For more information about the style-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src
+	StyleSrc DirectiveType = "style-src"
+	// font-src directive specifies valid sources for fonts loaded using @font-face.
+	// For more information about the font-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/font-src
+	FontSrc DirectiveType = "font-src"
+)
+
+// ConsolePluginCSP holds configuration for a specific CSP directive
+type ConsolePluginCSP struct {
+	// directive specifies which Content-Security-Policy directive to configure.
+	// Available directive types are default-src, script-src, img-src, style-src and font-src.
+	// +kubebuilder:validation:Enum:="default-src";"script-src";"img-src";"style-src";"font-src"
+	// +kubebuilder:validation:Required
+	Directive DirectiveType `json:"directive"`
+	// values defines an array of additional values to append to the console
+	// defaults for this directive.
```

Comment:
Don't really need additional do we?
```suggestion
	// values defines an array of values to append to the console
	// defaults for this directive.
```